pure_protobuf/dataclasses_.py

```diff
@@ -102,12 +83,22 @@ def loads(cls: Type[TMessage], bytes_: bytes) -> TMessage:
         return load(cls, io)
 
 
-def field(number: int, *args, packed=True, **kwargs) -> Any:
+def _field(number: int, *args, packed=True, isoneof=False, **kwargs) -> Any:
     """
     Convenience function to assign field numbers.
     Calls the standard ``dataclasses.field`` function with the metadata assigned.
     """
-    return dataclasses.field(*args, metadata={'number': number, 'packed': packed}, **kwargs)
+    metadata = {
+        'number': number,
+        'packed': packed,
+        'isoneof': isoneof
+    }
+
+    return dataclasses.field(*args, metadata=metadata, **kwargs)
+
+
+def field(number: int, *args, packed=True, **kwargs) -> Any:
+    return _field(number, *args, packed=packed, **kwargs)
```

Comment:
Could you elaborate on why this «proxy»-function is needed?

Reply:
I just needed to pass `isoneof` attribute to `metadata` (it can be dropped but better to use it than call `isinstance` function to find `OneOf_` fields later) and I intended here to reuse code of previous function `field` and not rewrite field init logic. 

I could pass `isoneof` into `field` itself but should we expose `isoneof` arg for usual fields in usual classes? I mean addition to list of args to previous func like:
```python 
def field(number: int, *args, packed=True, isoneof=False, **kwargs) -> Any:
```
seems to be a bad idea.

So I moved code a bit and renamed. Now it's invoked from `field` and later from `one_of`.

Mb should just copy and inline code of `_field` into body of `one_of`, don't know.

Reply:
Yeah, I've just overlooked the presence of `isoneof=False` which you hide for a purpose. Let's keep it like that